js/plugins/cloud-sql-pg/src/index.ts

```diff
@@ -127,26 +145,150 @@ export function configurePostgresRetriever<
  * @param ai A Genkit instance
  * @param params The params for the indexer
  * @param params.tableName The name of the indexer
+ * @param params.engine The engine to use for the indexer
  * @param params.embedder The embedder to use for the retriever
  * @param params.embedderOptions  Options to customize the embedder
- * @returns A Genkit indexer
+ * @param params.metadataColumns The metadata columns to use for the indexer
+ * @param params.idColumn The id column to use for the indexer
+ * @param params.metadataJsonColumn The metadata json column to use for the indexer
+ * @param params.distanceStrategy The distance strategy to use for the indexer
+ * @param params.contentColumn The content column to use for the indexer
+ * @param params.embeddingColumn The embedding column to use for the indexer
+ * @param params.schemaName The schema name to use for the indexer
+ * @param params.chunkSize The chunk size to use for the indexer
+ * @returns Add documents to vector store
  */
 export function configurePostgresIndexer<
   EmbedderCustomOptions extends z.ZodTypeAny,
 >(
   ai: Genkit,
   params: {
     tableName: string;
+    engine: PostgresEngine;
+    schemaName?: string;
+    contentColumn?: string;
+    embeddingColumn?: string;
+    metadataColumns?: string[];
+    idColumn?: string;
+    metadataJsonColumn?: string;
+    distanceStrategy?: 'cosine' | 'ip' | 'l2';
     embedder: EmbedderArgument<EmbedderCustomOptions>;
     embedderOptions?: z.infer<EmbedderCustomOptions>;
   }
 ) {
+  if (!params.engine) {
+    throw new Error('Engine is required');
+  }
+
+  const {
+    tableName,
+    engine,
+    schemaName = 'public',
+    contentColumn = 'content',
+    embeddingColumn = 'embedding',
+    metadataColumns = [],
+    idColumn = 'id',
+    metadataJsonColumn,
+    distanceStrategy = 'cosine',
+    embedder,
+    embedderOptions
+  } = params;
+
+  async function ensureTableExists() {
```

Comment:
The table will be created separately, but we should check that everything exists https://github.com/averikitsch/langchainjs/blob/6d78313ae24d020ffc69be3d4bf8d637eaa7f237/libs/langchain-google-cloud-sql-pg/src/vectorStore.ts#L114-L165